### ISSUE TITLE
type abbreviations cannot have the AllowNullLiteral attribute

### ISSUE THREAD

**Issue by ctaggart** [fable-compiler/ts2fable#14]
Experimenting with using gRPC communication from Fable. I ran into a couple of issues with the code gen.

```
ts2fable .\node_modules\@types\google-protobuf\index.d.ts > .\FableApp\src\google-protobuf.fs
```

![image](https://user-images.githubusercontent.com/80104/31454151-5e5d0be0-aeac-11e7-985c-b87e97f1792d.png)

https://github.com/ctaggart/grpc-web/blob/d11754b302daf0e90d6971d01e8e2f31a6f26e2a/example/FableApp/src/google-protobuf.fs

``` fsharp
namespace Fable.Import
open System
open System.Text.RegularExpressions
open Fable.Core
open Fable.Import.JS

type [<AllowNullLiteral>] ByteSource =
    U4<ArrayBuffer, Uint8Array, ResizeArray<float>, string>

and [<AllowNullLiteral>] ScalarFieldType =
    U3<bool, float, string>

and [<AllowNullLiteral>] RepeatedFieldType =
    U2<ResizeArray<ScalarFieldType>, ResizeArray<Uint8Array>>

and [<AllowNullLiteral>] AnyFieldType =
    U3<ScalarFieldType, RepeatedFieldType, Uint8Array>

and [<AllowNullLiteral>] FieldValue =
    obj

and [<AllowNullLiteral>] FieldValueArray =
    inherit ResizeArray<FieldValue>
```

**Comment by ctaggart** [comment]
These appear to be type abbreviations, so those can't type `[<AllowNullLiteral>]`. 

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/google-protobuf/index.d.ts#L6-L10
```
type ByteSource = ArrayBuffer | Uint8Array | number[] | string;
type ScalarFieldType = boolean | number | string;
type RepeatedFieldType = ScalarFieldType[] | Uint8Array[];
type AnyFieldType = ScalarFieldType | RepeatedFieldType | Uint8Array;
type FieldValue = string | number | boolean | Uint8Array | FieldValueArray | undefined;
```

From this file, I'll just remove it manually, but will rename this issue.

**Comment by alfonsogarciacaro** [comment]
You're right, we need to [remove `AllowNullLiteral` from the type declaration template](https://github.com/fable-compiler/ts2fable/blob/d5d88c74c4d06b0b9954299d664018b71771d745/ts2fable.js#L18) and add a check before including it :+1:

**Comment by ctaggart** [comment]
Fixed by #19.
